frontend/src/metabase-lib/queries/NativeQuery.ts

```diff
@@ -387,10 +389,18 @@ export default class NativeQuery extends AtomicQuery {
     return tagErrors.length === 0;
   }
 
-  setTemplateTag(name: string, tag: TemplateTag) {
-    return this.setDatasetQuery(
-      assocIn(this.datasetQuery(), ["native", "template-tags", name], tag),
+  setTemplateTag(name: string, tag: TemplateTag, config?: ParameterConfig) {
+    const newQuery = this.setDatasetQuery(
+      updateIn(this.datasetQuery(), ["native", "template-tags"], tags => ({
```

Comment:
Previously `template-tags` were updated by a redux action defined in `native.js` and not by this method. It turned out that it had issues and it will break the template tag order in the object. Here we fix this by using the same approach used in the action.